api/filters/check_instance_visibility_filter.go

```diff
@@ -100,8 +104,18 @@ func (f *serviceInstanceVisibilityFilter) Run(req *web.Request, next web.Handler
 	if len(visibility.PlatformID) == 0 { // public visibility
 		return next.Handle(req)
 	}
-	tenantLabels, ok := visibility.Labels[f.tenantIdentifier]
-	if ok && slice.StringsAnyEquals(tenantLabels, tenantID) {
+
+	if visibilityMetadata.LabelKey == "" {
+		return next.Handle(req)
+	}
+
+	tenantLabels, ok := visibility.Labels[visibilityMetadata.LabelKey]
+	if ok && slice.StringsAnyEquals(tenantLabels, visibilityMetadata.LabelValue) {
+		return next.Handle(req)
+	}
+
+	if !ok {
```

Comment:
Why is this not tested?
When finding bugs, we make sure to add tests for the new bug, so that we can be confident going forward that it doesn't reoccur.

Reply:
added test for this specific case - plan with public visibility but that supports only specific platform